packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -257,18 +257,13 @@ const App = (props) => {
     }
 
     const onAccountClick = () => {
-        // Link to account page for registered customer, open auth modal otherwise
-        if (isRegistered) {
-            const path = buildUrl('/account')
-            history.push(path)
-        } else {
-            // if they already are at the login page, do not show login modal
```

Comment:
This was a bug reported from UX awhile ago saying if a user is current in login page, the account click should not show the modal since user is already seeing the login form

Reply:
That bug is also handled by the `withRegistration` component. Although I noticed that here the logic handled `/login`, and the `withRegistration` component handles `/{locale}/login`, but what the app actually uses is `/global/{locale}/login`. I've changed the `withRegistration` component to be more flexible.